File: 1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.java
class Solution {
    public int[] frequencySort(int[] nums) {
        Map<Integer, Integer> freqMap = new HashMap<>();
        for (int num : nums) {
            freqMap.put(num, freqMap.getOrDefault(num, 0) + 1);
        }

        Integer[] numsBoxed = Arrays.stream(nums).boxed().toArray(Integer[]::new);

        Arrays.sort(numsBoxed, (a, b) -> {
            int freqA = freqMap.get(a);
            int freqB = freqMap.get(b);
            if (freqA == freqB) {
                return b - a;
            }
            return freqA - freqB;
        });

        return Arrays.stream(numsBoxed).mapToInt(Integer::intValue).toArray();
    }
}

File: LeetCode/1636-sort-array-by-increasing-frequency/1636-sort-array-by-increasing-frequency.cpp
class Solution {
public:
    vector<int> frequencySort(vector<int>& nums) {
        map<int, int> mp;
        for (int x : nums) {
            mp[x]++;
        }
        auto cmp = [&](int a, int b) {
            if (mp[a] == mp[b]) {
                return b < a;
            }
            return mp[a] < mp[b];
        };
        sort(nums.begin(), nums.end(), cmp);
        return nums;
    }
};